src/Utils.ts

```diff
@@ -919,42 +932,113 @@ export class DummyInputManager {
         const input = dummy.input;
 
         if (input) {
-            const parent = element.parentElement;
+            let parent: HTMLElement | null;
```

Comment:
Please try and rethink the code here, I can't understand what's going on here after multiple reads

Reply:
I have added a comment describing the phantom dummy concept.